src/material/progress-bar/progress-bar.scss

```diff
@@ -39,16 +31,306 @@
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  margin: auto 0;
+  width: 100%;
+  animation: none;
+  transform-origin: top left;
+  transition: transform 250ms 0ms cubic-bezier(0.4, 0, 0.6, 1);
+
+  @include token-utils.use-tokens(
+    tokens-mdc-linear-progress.$prefix,
+    tokens-mdc-linear-progress.get-token-slots()
+  ) {
+    @include token-utils.create-token-slot(height, active-indicator-height);
+  }
+
+  .mdc-linear-progress--indeterminate & {
+    transition: none;
+  }
+
+  [dir='rtl'] & {
+    right: 0;
+    transform-origin: center right;
+  }
+}
+
+.mdc-linear-progress__bar-inner {
+  display: inline-block;
+  position: absolute;
+  width: 100%;
+  animation: none;
+  border-top-style: solid;
+
+  @include token-utils.use-tokens(
+    tokens-mdc-linear-progress.$prefix,
+    tokens-mdc-linear-progress.get-token-slots()
+  ) {
+    @include token-utils.create-token-slot(border-color, active-indicator-color);
+    @include token-utils.create-token-slot(border-top-width, active-indicator-height);
+  }
+}
+
+.mdc-linear-progress__buffer {
+  display: flex;
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  margin: auto 0;
+  width: 100%;
+  overflow: hidden;
+
+  @include token-utils.use-tokens(
+    tokens-mdc-linear-progress.$prefix,
+    tokens-mdc-linear-progress.get-token-slots()
+  ) {
+    @include token-utils.create-token-slot(height, track-height);
+    @include token-utils.create-token-slot(border-radius, track-shape);
+  }
+}
+
+.mdc-linear-progress__buffer-dots {
+  $mask: "data:image/svg+xml,%3Csvg version='1.1' xmlns='http://www.w3.org/2000/svg' " +
+         "xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' " +
+         "enable-background='new 0 0 5 2' xml:space='preserve' viewBox='0 0 5 2' " +
+         "preserveAspectRatio='xMinYMin slice'%3E%3Ccircle cx='1' cy='1' r='1'/%3E%3C/svg%3E";
+
+  @include vendor-prefixes.mask-image(url($mask));
+  background-repeat: repeat-x;
+  flex: auto;
+  transform: rotate(180deg);
+  animation: mdc-linear-progress-buffering 250ms infinite linear;
+
+  @include token-utils.use-tokens(
+    tokens-mdc-linear-progress.$prefix,
+    tokens-mdc-linear-progress.get-token-slots()
+  ) {
+    @include token-utils.create-token-slot(background-color, track-color);
+  }
+
+  @include cdk.high-contrast(active, off) {
+    background-color: ButtonBorder;
+  }
+
+  [dir='rtl'] & {
+    animation: mdc-linear-progress-buffering-reverse 250ms infinite linear;
+    transform: rotate(0);
+  }
+}
+
+.mdc-linear-progress__buffer-bar {
+  flex: 0 1 100%;
+  transition: flex-basis 250ms 0ms cubic-bezier(0.4, 0, 0.6, 1);
+
+  @include token-utils.use-tokens(
+    tokens-mdc-linear-progress.$prefix,
+    tokens-mdc-linear-progress.get-token-slots()
+  ) {
+    @include token-utils.create-token-slot(background-color, track-color);
+  }
+}
+
+.mdc-linear-progress__primary-bar {
+  transform: scaleX(0);
+
+  .mdc-linear-progress--indeterminate & {
+    left: -145.166611%;
```

Comment:
where do these percentage values come from? Is it just what MDC used?

Reply:
Yep, they were basically hardcoded as CSS variables in MDC. I just inlined them since nobody is realistically changing those variables.